app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -31,4 +46,62 @@ class StateFragmentPresenter @Inject constructor(
   fun setAudioFragmentVisible(isVisible: Boolean) {
     getStateViewModel().setAudioFragmentVisible(isVisible)
   }
+
+  private fun createDummyData() {
+    interactionInstanceId = "ItemSelectionInput"
+    var sampleData: GaeCustomizationArgs =
+      GaeCustomizationArgs(true, "<p>The numerator.</p>, <p>The denominator.</p>, <p>I can't remember!</p>]")
+    gaeCustomizationArgsMap?.put("choices", sampleData)
+  }
+
+  private fun showInputInteractions(binding: StateFragmentBinding) {
+    val gaeCustomizationArgs: Any? = gaeCustomizationArgsMap!!.get("choices")?.value
+    if (interactionInstanceId.equals("MultipleChoiceInput")) {
+      val gaeCustomArgsInString: String = gaeCustomizationArgs.toString().replace("[", "").replace("]", "")
+      var items = gaeCustomArgsInString.split(",").toTypedArray()
+      addRadioButtons(items,binding)
+    } else if (interactionInstanceId.equals("ItemSelectionInput") || interactionInstanceId.equals("SingleChoiceInput")) {
+      val gaeCustomArgsInString: String = gaeCustomizationArgs.toString().replace("[", "").replace("]", "")
+      var items = gaeCustomArgsInString.split(",").toTypedArray()
+      addCheckbox(items,binding)
+    } else {
+      //Do no show any view
+    }
+  }
+
+  fun addCheckbox(optionsArray: Array<String>, binding: StateFragmentBinding) {
+    for (row in 0..0) {
+      for (i in 0..optionsArray.size - 1) {
+        val cb = CustomCheckbox(context, optionsArray[i])
+        binding.root.interactionContainer.addView(cb)
```

Comment:
We should be adding views via a RecyclerView, instead. I think that using a RecyclerView + data-binding on the custom items will make it much easier to bind content to the custom checkbox/radio button, and will be closer to the final solution for this fragment.

Reply:
@BenHenning So I shall create separate adapters for these views?

Reply:
Added views vai RecyclerView. Removed custom views for interactions and added it in Interaction Adapter